cartography/intel/aws/cloudtrail.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import boto3
+import botocore.exceptions
+import neo4j
+
+from cartography.client.core.tx import load
+from cartography.graph.job import GraphJob
+from cartography.intel.aws.ec2.util import get_botocore_config
+from cartography.models.aws.cloudtrail.trail import CloudTrailTrailSchema
+from cartography.util import aws_handle_regions
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_cloudtrail_trails(
+    boto3_session: boto3.Session, region: str
+) -> List[Dict[str, Any]]:
+    client = boto3_session.client(
+        "cloudtrail", region_name=region, config=get_botocore_config()
+    )
+    paginator = client.get_paginator("list_trails")
+    trails = []
+    for page in paginator.paginate():
+        trails.extend(page["Trails"])
+
+    # CloudTrail multi-region trails are shown in list_trails,
+    # but the get_trail call only works in the home region
+    trails_filtered = [trail for trail in trails if trail.get("HomeRegion") == region]
+    return trails_filtered
+
+
+@timeit
+def get_cloudtrail_trail(
```

Comment:
We probably should decorate this with `@aws_handle_regions` and then remove the `try + exception` here since the decorator does the exc handling. Not blocking though